### PR TITLE
Fix position problem to support more handheld devices with screen rotated

### PR DESCRIPTION
**Pull request by MiZai2**
I wrote this to fix position problem for handheld devices with screen rotated.

### CODE REVIEW

File: src/app/main.cpp [hunk 1]
```diff
@@ -289,12 +309,15 @@ int main(int, char**)
...
     sw_stats.deviceName = deviceName;
-    if (deviceName.find("Radeon") != std::string::npos
+    if (vendor.find("AMD") != std::string::npos
+    || deviceName.find("Radeon") != std::string::npos
     || deviceName.find("AMD") != std::string::npos){
         vendorID = 0x1002;
-    } else if (deviceName.find("Intel") != std::string::npos) {
+    } else if (vendor.find("Intel") != std::string::npos
+    || deviceName.find("Intel") != std::string::npos) {
         vendorID = 0x8086;
```
Comment (evelikov) [inline src/app/main.cpp:321]: The vendor detection changes should probably be a separate PR.

File: src/app/main.cpp [hunk 2]
```diff
@@ -245,11 +247,26 @@ static bool render(GLFWwindow* window) {
...
+    glfwSetWindowSize(window, screen_width, screen_height);
     ImGui::EndFrame();
     return last_window_size.x != window_size.x || last_window_size.y != window_size.y;
 }
 
+static void updateScreenInfo(){
+    GLFWmonitor* pMonitor = glfwGetPrimaryMonitor();
+    const GLFWvidmode * mode = glfwGetVideoMode(pMonitor);
+    screen_width = mode->width;
+    screen_height = mode->height;
```
Comment (evelikov) [inline src/app/main.cpp:259]: Will this work correctly when the primary (and only) monitor gets disconnected? Say when your dog pulls the cable for a moment, because they want you to play with them :dog2:

File: src/app/main.cpp [hunk 3]
```diff
@@ -289,12 +309,15 @@ int main(int, char**)
     notifier.params = &params;
     start_notifier(notifier);
     window_size = ImVec2(params.width, params.height);
+    std::string vendor = (char*)glGetString(GL_VENDOR);
```
Comment (MiZai2) [inline src/app/main.cpp]:
```suggestion
```

**Review by MiZai2 — COMMENTED** [review]
I wrote this to support more resolutions.
1.fix some problem cause by zero monitor since last commit.
2.recover vendor code.

File: src/app/main.cpp (hunk 2 above)
Comment (MiZai2) [reply src/app/main.cpp]: I will update a new file to resolve those problems
